plugin/pkg/scheduler/scheduler.go

```diff
@@ -107,6 +107,13 @@ func New(c *Config) *Scheduler {
 
 // Run begins watching and scheduling. It starts a goroutine and returns immediately.
 func (s *Scheduler) Run() {
+	go func() {
```

Comment:
can we move this to metrics.go and add a func called `ReportRateLoop` or something like that?